Failed to install using make build-all  (Mac OS High Sierra)

With up-to-date packages. Stack is "Version 1.7.1 x86_64".  
After doing make build-all on fresh git clone, I obtained the following output:
```
Downloading nightly-2018-07-26 build plan ...
Downloaded nightly-2018-07-26 build plan.
git submodule update --init
stack --stack-yaml=stack-8.2.1.yaml install happy
Downloaded nightly-2017-11-24 build plan.
Preparing to install GHC to an isolated location.
This will not interfere with any system-level installation.
Downloaded ghc-8.2.1.
Installed GHC.
[1 of 2] Compiling Main             ( /Users/hoge/.stack/setup-exe-src/setup-mPHDZzAJ.hs, /Users/hoge/.stack/setup-exe-src/setup-mPHDZzAJ.o )
[2 of 2] Compiling StackSetupShim   ( /Users/hoge/.stack/setup-exe-src/setup-shim-mPHDZzAJ.hs, /Users/hoge/.stack/setup-exe-src/setup-shim-mPHDZzAJ.o )
Linking /Users/hoge/.stack/setup-exe-cache/x86_64-osx/tmp-Cabal-simple_mPHDZzAJ_2.0.0.2_ghc-8.2.1 ...
clang: warning: argument unused during compilation: '-nopie' [-Wunused-command-line-argument]
mtl-2.2.1: download
mtl-2.2.1: configure
mtl-2.2.1: build
mtl-2.2.1: copy/register
happy-1.19.8: download
happy-1.19.8: configure
happy-1.19.8: build
happy-1.19.8: copy/register
Completed 2 action(s).
Copying from /Users/hoge/.stack/snapshots/x86_64-osx/nightly-2017-11-24/8.2.1/bin/happy to /Users/hoge/.local/bin/happy

Copied executables to /Users/hoge/.local/bin:
- happy

Warning: Installation path /Users/hoge/.local/bin
         not found on the PATH environment variable.
stack --stack-yaml=stack-8.2.1.yaml build

Error: While constructing the build plan, the following exceptions were encountered:

In the dependencies for cabal-helper-0.8.1.0:
    base-4.10.0.0 from stack configuration does not match <0 && <0
needed due to haskell-ide-engine-0.2.2.0 -> cabal-helper-0.8.1.0

Some different approaches to resolving this:

  * Set 'allow-newer: true' to ignore all version constraints and build anyway.

  * Consider trying 'stack solver', which uses the cabal-install solver to attempt to find
    some working build configuration. This can be convenient when dealing with many
    complicated constraint errors, but results may be unpredictable.


Plan construction failed.
make: *** [hie-8.2.1] Error 1
```

Same for me on macOS High Sierra 1.13.4:
```
davids-mbp:haskell davl3232$ cd haskell-ide-engine && make build-all
Downloading nightly-2018-07-26 build plan ...
Downloaded nightly-2018-07-26 build plan.
git submodule update --init
stack --stack-yaml=stack-8.2.1.yaml install happy
Downloaded nightly-2017-11-24 build plan.    
Preparing to install GHC to an isolated location.
This will not interfere with any system-level installation.
Downloaded ghc-8.2.1.                                      
Installed GHC.                                                                  [1 of 2] Compiling Main             ( /Users/davl3232/.stack/setup-exe-src/setup-mPHDZzAJ.hs, /Users/davl3232/.stack/setup-exe-src/setup-mPHDZzAJ.o )
[2 of 2] Compiling StackSetupShim   ( /Users/davl3232/.stack/setup-exe-src/setup-shim-mPHDZzAJ.hs, /Users/davl3232/.stack/setup-exe-src/setup-shim-mPHDZzAJ.o )
Linking /Users/davl3232/.stack/setup-exe-cache/x86_64-osx/tmp-Cabal-simple_mPHDZzAJ_2.0.0.2_ghc-8.2.1 ...
clang: warning: argument unused during compilation: '-nopie' [-Wunused-command-line-argument]
mtl-2.2.1: download
mtl-2.2.1: configure
mtl-2.2.1: build
mtl-2.2.1: copy/register
happy-1.19.8: download    
happy-1.19.8: configure   
happy-1.19.8: build       
happy-1.19.8: copy/register
Completed 2 action(s).    
Copying from /Users/davl3232/.stack/snapshots/x86_64-osx/nightly-2017-11-24/8.2.1/bin/happy to /Users/davl3232/.local/bin/happy

Copied executables to /Users/davl3232/.local/bin:
- happy

Warning: Installation path /Users/davl3232/.local/bin
         not found on the PATH environment variable.
stack --stack-yaml=stack-8.2.1.yaml build

Error: While constructing the build plan, the following exceptions were
encountered:

In the dependencies for cabal-helper-0.8.1.0:
    base-4.10.0.0 from stack configuration does not match <0 && <0 
needed due to haskell-ide-engine-0.2.2.0 -> cabal-helper-0.8.1.0

Some different approaches to resolving this:

  * Set 'allow-newer: true' to ignore all version constraints and build anyway.

  * Consider trying 'stack solver', which uses the cabal-install solver to
    attempt to find some working build configuration. This can be convenient
    when dealing with many complicated constraint errors, but results may be
    unpredictable.


Plan construction failed.
make: *** [hie-8.2.1] Error 1
```

--- a/stack-8.2.1.yaml
+++ b/stack-8.2.1.yaml
@@ -24,7 +24,7 @@ packages:
 extra-deps:
 - brittany-0.11.0.0
 - butcher-1.3.1.1
-- cabal-helper-0.8.1.0
+- cabal-helper-0.8.1.0@rev:0
 - cabal-plan-0.3.0.0
 - constrained-dynamic-0.1.0.0
 - czipwith-1.0.1.0

--- a/stack-8.2.2.yaml
+++ b/stack-8.2.2.yaml
@@ -24,7 +24,7 @@ packages:
 extra-deps:
 - brittany-0.11.0.0
 - butcher-1.3.1.1
-- cabal-helper-0.8.1.0
+- cabal-helper-0.8.1.0@rev:0
 - cabal-plan-0.3.0.0
 - conduit-parse-0.2.1.0
 - constrained-dynamic-0.1.0.0

--- a/stack-8.4.2.yaml
+++ b/stack-8.4.2.yaml
@@ -24,7 +24,7 @@ packages:
 extra-deps:
 - base-compat-0.9.3
 - brittany-0.11.0.0
-- cabal-helper-0.8.1.0
+- cabal-helper-0.8.1.0@rev:0
 - cabal-plan-0.3.0.0
 - constrained-dynamic-0.1.0.0
 - haddock-api-2.20.0

--- a/stack-8.4.3.yaml
+++ b/stack-8.4.3.yaml
@@ -23,7 +23,7 @@ packages:
 
 extra-deps:
 - base-compat-0.9.3
-- cabal-helper-0.8.1.0
+- cabal-helper-0.8.1.0@rev:0
 - cabal-plan-0.3.0.0
 - constrained-dynamic-0.1.0.0
 - haddock-api-2.20.0

--- a/stack.yaml
+++ b/stack.yaml
@@ -26,7 +26,7 @@ extra-deps:
 - apply-refact-0.5.0.0
 - base-compat-0.9.3
 - brittany-0.11.0.0
-- cabal-helper-0.8.1.0
+- cabal-helper-0.8.1.0@rev:0
 - cabal-plan-0.3.0.0
 - constrained-dynamic-0.1.0.0
 - ekg-0.4.0.15

